plugins/modules/infra-envs.py

```diff
@@ -0,0 +1,146 @@
+        default: null
+        type: str
+    name:
+        description: Name used to create a infra-envs resource. Note that this is required for infra-envs resource create operations.
+        required: false
+        type: str
+    pull_secret:
+        description: A secret object that stores credentials to pull container images from private registries. Note that is is required for create operations.
+        required: false
+        type: str
+
+author:
+    - Vishwanath Jayaraman (@vjayaramrh)
+
+"""
+
+EXAMPLES = r"""
+# Use argument
+- name: create infra-envs
+  infra-envs:
+    state: present
+    name: infra1
+    pull_secret: blah
+    
+- name: List infra-envs
+  infra-envs: {}
+  
+- name: Retrieve details of a specific infra-envs resource
+  infra-envs:
+    infra_env_id: "deadbeef-dead-beef-dead-beefdeadbeef"
+
+- name: Delete infra-envs
+  infra-envs:
+    state: absent
+    infra_env_id: "deadbeef-dead-beef-dead-beefdeadbeef"
+"""
+
+RETURN = r"""
+infra-envs:
+    description: A message with the status
+    type: list
+    returned: always
+    sample: []
+"""
+
+import os
+import traceback
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.basic import missing_required_lib
+
+try:
+    import requests
+except ImportError:
+    HAS_REQUESTS = False
+    REQUESTS_IMPORT_ERROR = traceback.format_exc()
+else:
+    HAS_REQUESTS = True
+    REQUESTS_IMPORT_ERROR = None
+
+API_VERSION = "v2"
+API_URL = f"https://api.openshift.com/api/assisted-install/{API_VERSION}"
+
+# add additional query parameters to the query_params_list
+QUERY_PARAMS_LIST = []
+
+
+def run_module():
+
+    module_args = dict(
+        state=dict(
+            type="str", required=False, choices=["absent", "present"], default=None
+        ),
+        # any API query parameters may have to be added here
+        name=dict(type="str", required=False),
+        pull_secret=dict(type="str", required=False),
+    )
+    token = os.environ.get("AI_API_TOKEN")
+    module = AnsibleModule(
+        argument_spec=module_args,
+        required_if=[
+            ("state", "present", ["name", "pull_secret"])
+        ],
+        supports_check_mode=False,
+    )
+
+    # Fail if requests is not installed
+    if not HAS_REQUESTS:
+        module.fail_json(
+            msg=missing_required_lib("requests"), exception=REQUESTS_IMPORT_ERROR
+        )
+
+    # Set headers
+    headers = {"Content-Type": "application/json", "Authorization": f"Bearer {token}"}
+
+    # Create infra-envs
+    if module.params.get("state") == "present":
+
+        data = remove_module_fields(module)
+
+        response = requests.post(f"{API_URL}/infra-envs", headers=headers, json=data)
+
+        if not response.ok:
+            result = dict(changed=True, response=response.text)
+            module.fail_json(msg="Error creating infra-envs", **result)
+
+        result = {"infra-envs": response.json()}
+
+    module.exit_json(**result)
```

Comment:
It doesn't look like list and delete operations are implemented.

Reply:
They will be as part of the corresponding issues #70 and #68